### PR TITLE
Neo2 layer 4 (mod4) lock

### PR DESCRIPTION
**Pull request by Matzkowsky**
This PR adds layer 4 (mod4) lock functionality for Neo2 keyboard layout.

Instructions: simply **press mod4** modifiers such as "grave_accent_and_tilde" or "right_command" **alone** to lock or unlock layer 4 of this nice keyboard layout.

The additional functionality is automatically included in presets:
- "Neo2 mod 3 and 4 keys (Apple keyboard)"
- "Neo2 mod 3 and 4 keys (Windows keyboard)"

@jgosmann please review and comment

### CODE REVIEW

**Comment by tekezo** [comment]
Thanks!

File: src/json/neo2.json.erb [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
-        <% def set_mod_4(b)
-            { "set_variable" => { "name" => "neo2_mod_4", "value" => b } }
+        <%
+        # halt defaults to 'false'
+        def set_mod_4(value, halt = false)
+            if (halt)
+                { "set_variable" => { "name" => "neo2_mod_4", "value" => value }, "halt" => true }
+            else
+                { "set_variable" => { "name" => "neo2_mod_4", "value" => value } }
+            end
```
Comment (jgosmann) [inline src/json/neo2.json.erb:11]: I'm not super familiar with ruby, but is this `if` required or could one just use
```
{ "set_variable" => { "name" => "neo2_mod_4", "value" => value }, "halt" => halt }
```
?

Comment (jgosmann) [reply src/json/neo2.json.erb:11]: Also is the "halt" variable actually used anywhere? It seems, the halt status is handled via "value" being `2`?

**Review by jgosmann — COMMENTED** [review]
Thank you for the contribution! Just had time to look over the changes and had a question regarding potential simplifications of the additions, but looks good overall.

File: src/json/neo2.json.erb (hunk 1 above)
Comment (MuhammedZakir) [reply src/json/neo2.json.erb:11]: `halt` is not "variable" like `value`, but a "flag" like `lazy` and `repeat`. It is used to cancel/stop events. Your changes will add `"halt": false` to config, but generally, you don't add those flags unless you need it. So `if` is better in this case.

See https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/to/halt/.

**Comment by christophebeling** [comment]
Is there some opportunity to deactivate this feature? I'm pretty much used to the none locking layer 4.

**Comment by Matzkowsky** [comment]
> Is there some opportunity to deactivate this feature? I'm pretty much used to the none locking layer 4.

Hi @christophebeling . I didn't found a way to make this option configurable as for example with "Toggle caps_lock by pressing left_shift + right_shift at the same time". However I understand that some users don't like mod4 lock by double press mod4 key. One possible approach would be to have have 2 "Apple keyboard" layouts, 1 with, 1 without lock. Would that be an acceptable solution?

**Comment by christophebeling** [comment]
@Matzkowsky For me, it even locks with a single press. The solution with the 2 layouts would be fine though.

**Comment by david0** [comment]
Are these fixes already available via https://ke-complex-modifications.pqrs.org/? 
I deleted all my rules and imported them via "Import rules from the internet" but I have still the problem of a locking to layer 4 when I press mod4.

**Comment by jgosmann** [comment]
They should be available. Deleting the enabled rules, re-importing the rules, and enabling them again should work ...

**Comment by Matzkowsky** [comment]
I just checked and these changes are not updated in the rules portal. In the meantime you could unpack the attached archive and place the json file under ~/.config/karabiner/assets/complex_modifications/
[neo2.json.zip](https://github.com/pqrs-org/KE-complex_modifications/files/5601956/neo2.json.zip)

**Comment by jgosmann** [comment]
Indeed, the [deployment failed](https://github.com/pqrs-org/KE-complex_modifications/runs/1438517877). Not sure why or I am even able to do anything about that.

**Comment by jgosmann** [comment]
Opened #781 to get this fixed.

**Comment by jgosmann** [comment]
#781 got merged, seems like the changes are now available in the portal.

**Pull request merged**